src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+import { useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import './Autocomplete.scss';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [focusOnInput, setFocusOnInput] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [query, setQuery] = useState('');
+
+  const applyQuery = debounce(setAppliedQuery, delay);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+    applyQuery(value);
+    setFocusOnInput(true);
+    onSelected(null);
+  };
+
+  const handleSelectPerson = (person: Person) => {
+    onSelected(person);
+    setFocusOnInput(false);
+    setQuery(person.name);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person =>
+      person.name
```

Comment:
You are directly interacting with DOM here. It's recommended to use React as much as possible. You can use the `trim()` function when setting the state of `appliedQuery` instead of using it here.